### ISSUE TITLE
event.addMetadata incorrectly adds the same key multiple times

### ISSUE THREAD

**Issue by cristianoccazinsp** [bugsnag/bugsnag-js#1041]
### Description

I'm adding extra info to the metadata object (with `addMetadata`), but for some reason json keys are being nested and added twice. This didn't happen before the migration to `@bugsnag/react-native: 7.3.3`

### Issue

The following code is used in the `onError` callback to add extra data:

![image](https://user-images.githubusercontent.com/48869228/92308166-11abbf80-ef72-11ea-8308-f793018d4cad.png)

However, when the log is received, some keys are nested and duplicated into other keys, like this:

![image](https://user-images.githubusercontent.com/48869228/92308172-2c7e3400-ef72-11ea-93bf-d241e4f0d5f8.png)

This gets even worse the more objects are added to the log, ending up with massive payloads.

### Environment

Library versions:

```shell
@bugsnag/react-native: 7.3.3
react-native: 0.62.2
```

- iOS/Android version(s): All 
- simulator/emulator or physical device?: Device
- debug mode or production?: Prod

- [X] (iOS only) `[BugsnagReactNative start]` is present in the
  `application:didFinishLaunchingWithOptions:` method in your `AppDelegate`
  class?
- [X] (Android only) `BugsnagReactNative.start(this)` is present in the
  `onCreate` method of your `MainApplication` class?

### Example Repo

- [X] Create a minimal repository that can reproduce the issue after running
  `yarn install` and `react-native run-ios`/`react-native run-android`
- [X] Link to it here:

### Example code snippet

```
bugsnagConf.onError = function(event) {
  if (conf.env === 'dev'){
    console.warn("Bugsnag error not sent: ", event, event.errors);
    return false;
  }
  else{

    if (global.store){
      let state = global.store.getState();
      if(!state.settings.autoSendErrorLogs){
        return false;
      }

      let queueInfo =  {
        queue: {
          total: String(state.queue.total),
          error: String(state.queue.errors)
        },
        sync: {
          status: state.sync.status,
          statusInfo: state.sync.statusInfo,
          lastSync: state.sync.lastSync ? (new Date(state.sync.lastSync).toISOString()) : state.sync.lastSync,
          lastUpdate: String(state.sync.lastUpdate),
          currentTime: new Date().toISOString()
        },
        network: JSON.stringify(state.network)
      };

      event.addMetadata('queueInfo', queueInfo);

      return true;
    }
  }
};
```

I would love to know if this is an usage issue or an actual bug.  Either way, it is increasing data usage significantly.

**Comment by johnkiely1** [comment]
This has been transferred to the bugsnag-js repo as the Bugsnag react native library is now part of this repository (as of v7.3).

**Comment by johnkiely1** [comment]
Hi @cristianoccazinsp,

This has now been fixed and is released in [v7.4.0](https://github.com/bugsnag/bugsnag-js/releases/tag/v7.4.0).
